examples/binarySearch/binarySearch.act

```diff
@@ -0,0 +1,40 @@
+behaviour init of BinarySearch
+interface constructor()
+
+creates
+	uint256[] array := []
+
+invariants
+	sorted(array)
+
+where
+
+sorted (uint256[] array)
+case [] = true
+case [x] = true
+case [x,y,xs] = x <= y and sorted([y,xs])
```

Comment:
how about:
```
sorted : uint256[] -> bool
sorted ([]) = true
sorted ([x]) = true
sorted ([x;y;xs]) = x <= y and sorted([y;xs])
```

Reply:
possibly without the parens

Reply:
sounds good too

Reply:
why `;` instead of `,`?

Reply:
I wanted to distinguish between `(,) : a -> a -> [a]` and `(;) -> a -> [a] -> [a]`.
Another choice is `::`, which idris uses for example

Reply:
coq uses `;`. I think I prefer `::`

Reply:
updated